src/config.c

```diff
@@ -2490,6 +2490,27 @@ static int updatePort(const char **err) {
     return 1;
 }
 
+static int updateDatabases(const char **err) {
+    int old_dbnum = server.dbnum;
+    int new_dbnum = server.cluster_enabled ? server.config_databases_cluster : server.config_databases;
+    if (old_dbnum == new_dbnum) return 1;
+
+    /* If it is adjusted smaller, then we need to check whether the
+     * database has been used. */
+    if (new_dbnum < old_dbnum) {
+        for (int dbid = new_dbnum; dbid < old_dbnum; dbid++) {
+            if (server.db[dbid] != NULL) {
+                *err = "The database has been used and cannot be adjusted "
+                       "to a smaller number.";
+                return 0;
+            }
+        }
+    }
+    server.dbnum = new_dbnum;
+    server.db = zrealloc(server.db, sizeof(serverDb *) * server.dbnum);
```

Comment:
I have 2 concerns for this PR：
1. With which case, we need to update this number?
2.  If new_dbnum < old_dbnum, why we need to allocate new memory to server.db?

Reply:
1. we want to update the number to use more DBs without other process
2. both are ok, doing realloc look clearer to me